pkg/operator/ceph/config/conditions.go

```diff
@@ -171,6 +177,15 @@ func conditionMapping(conditions []cephv1.Condition) {
 	}
 }
 
+// getCluster returns the CephCluster
+func getCephCluster(context *clusterd.Context, namespace, name string) *cephv1.CephCluster {
+	cluster, err := context.RookClientset.CephV1().CephClusters(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		logger.Errorf("failed to get cluster %v", err)
```

Comment:
If this fails the cluster will be nil and could cause the caller to crash. How about if you don't have this helper method? It's very simple and only called in two places anyway. Then the callers can handle errors directly.

Reply:
There are some other instances too, but less than 5 I guess. And I thought I will return all the errors through another one PR I was working on https://github.com/rook/rook/pull/4894. What do you think? @travisn

Reply:
Either way, we need to handle the errors before merging this PR.